jre_emul/Classes/com/google/j2objc/security/IosRSAKey.java

```diff
@@ -343,5 +307,51 @@ protected void decodeParameters() {
         throw new AssertionError("failed decoding key parameters: " + e);
       }
     }
+    /**
+     * This method adds a private key to the keychain and returns the key-reference
+     * The format of the bytes are apply format.
+     * 
+     * @param bytes
+     * @return
+     */
+    private native long createPrivatSecKeyRef(byte[] bytes) /*-[
+    NSData * privateKey = [[NSData alloc] initWithBytes:(const void *)(bytes->buffer_)
+                                               length:bytes->size_];
+
+    // Delete any previous key definition.
+    NSMutableDictionary *keyQuery = getPrivateQuery();
+    OSStatus status = SecItemDelete((CFDictionaryRef) keyQuery);
+    if (status != errSecSuccess && status != errSecItemNotFound) {
+      NSString *msg = [NSString stringWithFormat:
+          @"Problem removing previous public key from the keychain, OSStatus == %d",
+          (int)status];
+      @throw create_JavaSecurityProviderException_initWithNSString_(msg);
+    }
+
+    // Store key in keychain.
+
+    keyQuery[(id)kSecAttrCanDecrypt] = (id)kCFBooleanTrue;
+    keyQuery[(id)kSecAttrCanDerive] = (id)kCFBooleanFalse;
```

Comment:
Why false? I don't understand crypto key use for derivation, but it's the Apple default.